### ISSUE TITLE
Accessibility audit

### ISSUE THREAD

**Issue by sra405** [RaspberryPiFoundation/editor-ui#1123]
## Accessibility audit for web-component.html

Axe accessibility audit to identify critical and non-critical (nice-to-have) issues. The audit output will be used to create backlog tickets for improvement.

### Focus Page
The web-component page is the primary focus for this audit, as it is the most critical part of the site.

- [ ] -  /web-component.html
- [ ] - sidebar options
### Output

A prioritized list of critical issues to address immediately.
A secondary list of easy-to-fix, non-critical issues for future improvements.

### Reference
Please review the findings from the [previous accessibility audit](https://github.com/RaspberryPiFoundation/projects-ui/issues/2274) for context and historical data.

**Comment by magdalenajadach** [comment]
## Output of the tests

### General
1. Lack of language attribute 
- Description: `<html> element does not have a [lang] attribute`

2. Modal focus issue
- Keyboard focus is not maintained within the modal. It is possible to tab out of the modal.
https://docs.deque.com/issue-help/1.0.0/en/focus-modal-moves-outside

3. Images
- aria-hidden="true" is used on informative content making it inaccessible to assistive technology.

<img width="1340" alt="Image" src="https://github.com/user-attachments/assets/2cd93e43-543f-436e-94a4-97297c236eb2" />
